src/main/kotlin/io/emeraldpay/dshackle/upstream/rpcclient/stream/JsonRpcStreamParser.kt

```diff
@@ -0,0 +1,243 @@
+package io.emeraldpay.dshackle.upstream.rpcclient.stream
+
+import com.fasterxml.jackson.core.JsonFactory
+import com.fasterxml.jackson.core.JsonToken
+import io.emeraldpay.dshackle.upstream.rpcclient.JsonRpcError
+import io.emeraldpay.dshackle.upstream.rpcclient.ResponseRpcParser
+import io.emeraldpay.etherjar.rpc.RpcResponseError
+import reactor.core.publisher.Flux
+import reactor.core.publisher.Mono
+import reactor.netty.ByteBufFlux
+import java.util.Arrays
+import java.util.concurrent.atomic.AtomicBoolean
+import java.util.concurrent.atomic.AtomicInteger
+import java.util.concurrent.atomic.AtomicReference
+
+class JsonRpcStreamParser {
+    companion object {
+        private val jsonFactory = JsonFactory()
+        private val responseRpcParser = ResponseRpcParser()
+
+        private const val ARRAY_OPEN_BRACKET: Byte = 91
```

Comment:
Maybe use smth like `'['.code.toByte()` for better readability ?

Reply:
Cool idea, done, thanks!